public/docs/advanced/commands.md

```diff
@@ -1,129 +1,215 @@
 # Command reference
 
+This section will describe every command available in the Truffle application.
+
+
 ## Usage
 
-```none
-$ truffle [command] [options]
+All commands are in the following form:
+
+```shell
+truffle [command] [options]
 ```
 
+No arguments given is equivalent to `truffle help`, which will display a list of all commands and then exit.
+
+
 ## Available commands
 
+
 ### build
 
-Execute build pipeline (if configuration present)
+Execute build pipeline (if configuration present).
 
-```none
-$ truffle build
+```shell
+truffle build
 ```
 
-See the [Building your application](/docs/getting_started/build) section for more details.
+Requires the `build` key to be present in the configuration. See the [Building your application](/docs/getting_started/build) and [Build processes](/docs/advanced/build_processes) sections for more details.
+
+
+### compile
+
+Compile contract source files.
+
+```shell
+truffle compile [--all] [--network <name>]
+```
+
+This will only compile contracts that have changed since the last compile, unless otherwise specified.
+
+Optional parameters:
+
+* `--all`: Compile all contracts instead of only the contracts changed since last compile.
+* `--network <name>`: Specify the network to use, saving artifacts specific to that network. Network name must exist in the configuration.
+
 
 ### console
 
 Run a console with contract abstractions and commands available.
 
-```none
-$ truffle console
+```shell
+truffle console [--network <name>|--verbose-rpc]
 ```
 
-Once the console starts, you can then use your contracts via the command line like you would in your code. Additionally, all truffle commands listed here are available within the console.
+Spawns an interface to interact with contracts via the command line. Additionally, many Truffle commands are additionally available within the console (without the `truffle` prefix).
+
+Requires an external RPC, such as [Ganache](/docs/ganache/using) or geth. For the console that generates its own RPC, use `truffle develop`.
+
+See the [Using the console](/docs/getting_started/console) section for more details.
 
 Optional parameters:
 
-* `--network name`: Specify the network to use.
+* `--network <name>`: Specify the network to use. Network name must exist in the configuration.
 * `--verbose-rpc`: Log communication between Truffle and the RPC.
```

Comment:
Should probably change RPC here too to something fancy.

Reply:
Might as well use "Ethereum client", as that's what we subbed for RPC above.

The command name itself (`--verbose-rpc`) might also be worth revisiting.